packages/common/src/major2-validation.ts

```diff
@@ -479,24 +514,16 @@ function validateAndRequirement(
   r: IAndCourse2,
   tracker: CourseValidationTracker
 ): Result<Array<Solution>, MajorValidationError> {
-  const splitResults = validateAndSplit(r.courses, tracker);
-  const [allChildReqSolutions, childErrors] = splitResults;
-
-  // AND's children has errors
-  if (childErrors.length > 0) {
-    return Err(AndErrorUnsatChild(childErrors));
-  }
+  const results = validateRequirements(r.courses, tracker);
+  const [allChildReqSolutions, childErrors] = splitChildResults(results);
```

Comment:
Validate and split separately so we can keep track of the original indices of sections for NoSolution index accuracy.

Reply:
a little unsure of why we need to do this, but will take a look at it again to better understand

Reply:
Basically we need to keep track of the index of each solution/error so that we can return the correct index for a no solution (or and unsat and no solution) error. We rely on the index in the UI to uncheck the right section. `splitChildResults` by itself returns a list of solutions and a list of errors, which are individually ordered, but we lose their position in the overall order, so we now keep track of the original order instead of just validating children and splitting in one step.